opennms-dao/src/main/java/org/opennms/netmgt/dao/hibernate/InterfaceToNodeCacheDaoImpl.java

```diff
@@ -358,6 +361,16 @@ public boolean removeNodeId(final String location, final InetAddress address, fi
             LOG.warn("removeNodeId: null IP address");
             return false;
         }
+        // can't infer location when node is deleted, delete all the interfaces with the node.
+        if (location == null) {
+            List<Map.Entry<Key, Value>> keyValues = m_managedAddresses.entries().stream()
+                    .filter(keyValueEntry -> keyValueEntry.getValue().getNodeId() == nodeId).collect(Collectors.toList());
```

Comment:
Shouldn't  you delete only entries where the node ID _and_ the IP address match?  This is currently matching _every_ entry in the cache that matches the node ID.

Reply:
Yeah, after further thought I think I will leave `removeNodeId` and introduce `removeInterfacesForNode` for `uei.opennms.org/nodes/nodeDeleted`